### PR TITLE
Fix CDS/CDNSKEY RRsets comparison in DNSSEC15

### LINKED ISSUE #1381: CDS/CDNSKEY inconsistency error
```
$ zonemaster-cli --version
Zonemaster-CLI version v7.0.0
Zonemaster-Engine version v6.0.0
Zonemaster-LDNS version 4.0.2
NL NetLabs LDNS version 1.8.3

$ zonemaster-cli 8.1.7.0.1.0.0.2.ip6.arpa --ns decsys.vsb.cz --ns nsa.ces.net --ns nsa.cesnet.cz
Loading profile from /etc/zonemaster/ripencc.json.
  /
Seconds Level    Message
======= ======== =======
   8.66 ERROR    All servers do not have the same CDS RRset.
   8.66 ERROR    All servers do not have the same CDNSKEY RRset.
```
I've removed the NOTICE and WARNING level messages. The interesting one is this ERROR about the CDS/CDNSKEY. I cannot see the inconsistency:
```
% dig +norec +short 8.1.7.0.1.0.0.2.ip6.arpa cds @nsa.cesnet.cz
37015 13 2 DCDB214392D046333752C10124170645EA86D299749F7953D66F8598 CE80CDD7
% dig +norec +short 8.1.7.0.1.0.0.2.ip6.arpa cds @nsa.ces.net
37015 13 2 DCDB214392D046333752C10124170645EA86D299749F7953D66F8598 CE80CDD7
% dig +norec +short 8.1.7.0.1.0.0.2.ip6.arpa cds @decsys.vsb.cz
37015 13 2 DCDB214392D046333752C10124170645EA86D299749F7953D66F8598 CE80CDD7
```

### PR DESCRIPTION
**Pull request by tgreenx**
## Purpose

This PR proposes a fix to the handling of CDS and CDNSKEY RRsets comparison in DNSSEC15.

PS: The proper fix for the code below https://github.com/zonemaster/zonemaster-engine/pull/1383/files#diff-3c6ad87ddb974fb813839bacdaafd4e212c238d0ff6865a9bf6cf677403343deR3953 requires changes in LDNS. PRs have been made to make it happen, and it should be available in the next release of LDNS (1.8.5). See https://github.com/NLnetLabs/ldns/commit/b39813870a5fb0f4e8ff1570b3b09416aaee716c.

## Context

Fixes #1381 

Requires https://github.com/zonemaster/zonemaster-ldns/pull/199

## Changes

- Update code logic
- Refactoring

## How to test this PR

_This requires https://github.com/zonemaster/zonemaster-ldns/pull/199._

Unit tests should pass.

Manual testing (from #1381):
```
$ zonemaster-cli 8.1.7.0.1.0.0.2.ip6.arpa --ns decsys.vsb.cz --ns nsa.ces.net --ns nsa.cesnet.cz --show-testcase --test dnssec15
  -
Seconds Level    Testcase       Message
======= ======== ============== =======
Looks OK.
```

### CODE REVIEW

**Review by matsduf — APPROVED** [review]
I find the code quite hard to follow. I would like to see comments that says what sections are meant to do. This is not due to the current PR. That is how the code is already. But that fact makes it harder to review the code.

I have tested the update and I have found no issues.

File: lib/Zonemaster/Engine/Test/DNSSEC.pm [hunk 1]
```diff
@@ -4004,42 +4002,45 @@ sub dnssec15 {
...
-                $first_rrset_string = $rrset_string;
+            if ( $first ) {
+                $first_rrlist = scalar @{ $cds_rrsets{ $ns_ip } } ? Zonemaster::LDNS::RRList->new( $cds_rrsets{ $ns_ip } ) : undef;
+                $first = 0;
+                next;
             }
-            elsif ( $rrset_string ne $first_rrset_string ) {
-                push @results, _emit_log( DS15_INCONSISTENT_CDS => {} );
+
+            my $rrlist = scalar @{ $cds_rrsets{ $ns_ip } } ? Zonemaster::LDNS::RRList->new( $cds_rrsets{ $ns_ip } ) : undef;
```
Comment (marc-vanderwal) [inline lib/Zonemaster/Engine/Test/DNSSEC.pm:4015]: As soon as https://github.com/zonemaster/zonemaster-ldns/pull/209 is merged, this can be simplified to:
```suggestion
            my $rrlist = Zonemaster::LDNS::RRList->new( $cds_rrsets{ $ns_ip } );
```

File: lib/Zonemaster/Engine/Test/DNSSEC.pm [hunk 2]
```diff
@@ -4004,42 +4002,45 @@ sub dnssec15 {
...
-            if ( scalar @{ $cds_rrsets{ $ns_ip } } ) {
-                $rrset_string = join( "\n", sort map { $_->string } @{ $cds_rrsets{ $ns_ip } } );
-            }
-            else {
-                $rrset_string = q{};
-            }
-            if ( not defined $first_rrset_string ) {
-                $first_rrset_string = $rrset_string;
+            if ( $first ) {
+                $first_rrlist = scalar @{ $cds_rrsets{ $ns_ip } } ? Zonemaster::LDNS::RRList->new( $cds_rrsets{ $ns_ip } ) : undef;
```
Comment (marc-vanderwal) [inline lib/Zonemaster/Engine/Test/DNSSEC.pm:4010]: As soon as https://github.com/zonemaster/zonemaster-ldns/pull/209 is merged, this can be simplified to:
```suggestion
                $first_rrlist = Zonemaster::LDNS::RRList->new( $cds_rrsets{ $ns_ip } );
```

**Review by marc-vanderwal — COMMENTED** [review]
I suggest that we wait until https://github.com/zonemaster/zonemaster-ldns/pull/209 is merged.

**Comment by matsduf** [comment]
Reviewed for release testing of v2024.2: no issues.
